test/cloudformation/cloudformation_test.go

```diff
@@ -0,0 +1,98 @@
+package cloudformation
+
+import (
+	"context"
+	"flag"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	timeOutMinutes = 15
+	instanceIdKey  = "InstanceId"
+	namespace      = "CFMetricTest"
+)
+
+var (
+	dimensionProviders = []dimension.IProvider{
+		&dimension.CustomDimensionProvider{Provider: dimension.Provider{}},
+	}
+	dimensionFactory = dimension.Factory{Providers: dimensionProviders}
+	packagePath      = flag.String("package_path", "", "s3 path to download package")
+	iamRole          = flag.String("iam_role", "", "iam role for cf ec2 instance")
+	keyName          = flag.String("key_name", "", "key name for ec2 instance")
+	metricName       = flag.String("metric_name", "", "metric to look for")
+)
+
+func TestCloudformation(t *testing.T) {
+	log.Printf("Package path %s iam role %s key name %s metric name %s", *packagePath, *iamRole, *keyName, *metricName)
+	ctx := context.Background()
+	stackName := awsservice.CreateStackName("cfTestStack")
+
+	client := awsservice.CloudformationClient
+	defer awsservice.DeleteStack(ctx, stackName, client)
```

Comment:
```
defer awsservice.DeleteStack(ctx, stackName, client)
```
should be called just after:
```
awsservice.StartStack(ctx, stackName, client, templateText, timeOutMinutes, parameters)
```

Probably not a big deal, but it is a good habit.